packages/core/src/StateNode.ts

```diff
@@ -1307,11 +1307,16 @@ class StateNode<
         action
       ): action is
         | RaiseActionObject<TEvent>
-        | SendActionObject<TContext, TEvent, TEvent> =>
-        action.type === actionTypes.raise ||
-        (action.type === actionTypes.send &&
+        | SendActionObject<TContext, TEvent, TEvent> => {
+        const isSendAction = action.type === actionTypes.send;
+        const isRaisableSendAction =
+          isSendAction &&
           (action as SendActionObject<TContext, TEvent>).to ===
-            SpecialTargets.Internal)
+            SpecialTargets.Internal &&
+          !(action as SendActionObject<TContext, TEvent>).delay;
```

Comment:
q: what should happen with `delay: 0`? 😉 is it raisable? or should it execute with `setTimeout(() => this.sendTo(event), 0)`?

Reply:
The latter; don't special-case it

Reply:
it would be cool to have a test for that case

Reply:
Done here: [b583f05](https://github.com/statelyai/xstate/pull/3393/commits/b583f05b58076830041eb0d50eaaa88f9ca2dcab)